app/code/Magento/MediaGallerySynchronization/Model/Filesystem/FileInfo.php

```diff
@@ -0,0 +1,251 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGallerySynchronization\Model\Filesystem;
+
+/**
+ * Internal class wrapping \SplFileInfo
+ *
+ * @SuppressWarnings(PHPMD.TooManyFields)
+ */
+class FileInfo extends \SplFileInfo
```

Comment:
It would be better to use ccomposition instead of inheritance

Reply:
This class does not need to extend `\SplFileInfo` please remove the inheritance